addons/website_sale/controllers/main.py

```diff
@@ -434,6 +437,10 @@ def cart_update(self, product_id, add_qty=1, set_qty=0, **kw):
             product_custom_attribute_values=product_custom_attribute_values,
             no_variant_attribute_values=no_variant_attribute_values
         )
+
+        if kw.get('express'):
+            return request.redirect("/shop/cart?express=1")
```

Comment:
why not directly return to return request.redirect("/shop/checkout?express=1") and remove redirect in def cart() ?